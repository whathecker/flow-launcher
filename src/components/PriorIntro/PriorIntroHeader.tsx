import React from "react";
import { StyleSheet, Image } from "react-native";
import { View, Text } from "../Themed";
import { CloseIcon } from "../shared";
import { Container, Typography } from "../../styles";
import { navigationRef } from "../../utils";

const PriorIntroHeader: React.FC = () => {
  return (
    <>
      <View style={styles.closeIconAreaWrapper}>
        <CloseIcon pressHandler={() => navigationRef.goBack()} />
      </View>
      <View style={styles.headerAreaWrapper}>
        <Image
          style={styles.headerImage}
          source={require(`../../../assets/images/bar-chart_1f4ca.png`)}
        />
        <Text style={styles.headerText}>{`Letâ€™s Prioritize Tasks`}</Text>
        <Text
          style={styles.subHeaderText}
        >{`We will ask two questions for each task`}</Text>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  closeIconAreaWrapper: {
    ...Container.flexStart,
    justifyContent: "flex-start",
    width: "90%",
    height: "50%",
    marginLeft: "4.5%",
    marginRight: "5%",
  },
  headerAreaWrapper: {
    ...Container.centerAlignedVertical,
  },
  headerImage: {
    width: 50,
    height: 50,
  },
  headerText: {
    ...Typography.h4,
    fontSize: 24,
    paddingTop: 10,
    paddingBottom: 5,
  },
  subHeaderText: {
    ...Typography.p,
    fontSize: 14,
  },
});

export default PriorIntroHeader;
