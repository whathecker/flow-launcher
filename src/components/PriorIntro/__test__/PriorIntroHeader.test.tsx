import * as React from "react";
import renderer from "react-test-renderer";
import PriorIntroHeader from "../PriorIntroHeader";

describe("Test PriorIntroHeader component", () => {
  it("Should renders PriorIntroHeader correctly", () => {
    const tree = renderer.create(<PriorIntroHeader />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
