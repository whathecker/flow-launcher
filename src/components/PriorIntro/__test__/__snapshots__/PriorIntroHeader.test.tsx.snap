// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PriorIntroHeader component Should renders PriorIntroHeader correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "50%",
          "justifyContent": "flex-start",
          "marginLeft": "5%",
          "marginRight": "5%",
          "width": "90%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "fontFamily": "EB-Garamond",
            "fontSize": 25,
            "fontWeight": "bold",
            "paddingBottom": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      Letâ€™s Prioritize Tasks
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "fontFamily": "Forum-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      We will ask two questions for each task
    </Text>
  </View>,
]
`;
